query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -406,7 +406,7 @@ abstract class NonLeafExecPlan extends ExecPlan {
       case QueryResult(_, schema, _) if rs == ResultSchema.empty =>
         schema     /// First schema, take as is
       case QueryResult(_, schema, _) =>
-        if (rs != schema) throw SchemaMismatch(rs.toString, schema.toString)
+        if (rs.hasSameColumnsAs(schema)) throw SchemaMismatch(rs.toString, schema.toString)
```

Comment:
This is tricky and needs more thought. 

Does this mean we can join counters and gauges ?

Reply:
Can we make this apply this fix only to joins ?
Doing it here, will apply for dist-concat as well...

Reply:
ack, fixed